src/pdb.rs

```diff
@@ -7,26 +8,22 @@ use std::io::Cursor;
 pub fn get_compact_symbol_table(
     pdb_data: &[u8],
     breakpad_id: &str,
-) -> Option<CompactSymbolTable> {
+) -> CompactSymbolResult<CompactSymbolTable> {
     // Now, parse the PDB and check it against the expected breakpad_id.
     let pdb_reader = Cursor::new(pdb_data);
-    let mut pdb = PDB::open(pdb_reader).ok()?;
-    let info = pdb.pdb_information().ok()?;
-    let pdb_id = format!(
-        "{}{:x}",
-        format!("{:X}", info.guid.to_simple()),
-        info.age
-    );
+    let mut pdb = PDB::open(pdb_reader)?;
+    let info = pdb.pdb_information()?;
+    let pdb_id = format!("{}{:x}", format!("{:X}", info.guid.to_simple()), info.age);
 
     if pdb_id != breakpad_id {
-        return None;
+        return Err(CompactSymbolError::UnmatchedString);
```

Comment:
When you have renamed this error variant to `UnmatchedBreakpadID`, I think it would be a good idea to store both breakpad IDs in it, so that the error can be displayed as "expected ID ..., found ID ...".